src/atomicassets.cpp

```diff
@@ -503,6 +503,43 @@ ACTION atomicassets::locktemplate(
 }
 
 
+/**
+* Reduces the max supply of the template to the new max supply
+* This means that afterwards, NFTs can only be minted up to the new max supply
+* @required_auth authorized_editor, who is within the authorized_accounts list of the collection
+**/
+
+ACTION atomicassets::redmaxtemplt(
+    name authorized_editor,
+    name collection_name,
+    int32_t template_id,
+    uint32_t new_max_supply
+) {
+    require_auth(authorized_editor);
+
+    auto collection_itr = collections.require_find(collection_name.value,
```

Comment:
This check is already done in check_has_collection_auth

Reply:
This is a good point, the check is redundant and can be removed here. It can actually also be removed from a number of other actions. I would suggest an additional PR removing the unnecessary checks.

Reply:
This is addressed in dev feature 8, which uses a new form of the  check_has_collection_auth & will be used to adjust all other pr’s